### ISSUE TITLE
How to sample from Gamma VAE?

### ISSUE THREAD

**Issue by AntixK** [darleybarreto/vae-pytorch#1]
Hi, 
Could you kindly point out how to sample images from the trained Gamma VAE model?
In the Gaussian case, we simply sample z from N(0, 1) and pass it through the decoder.

I did try sampling from the prior Gamma(2, 1) but the samples look horrible, albeit having decent reconstruction.

Any help is appreciated!

**Comment by darleybarreto** [comment]
Hi Anand, thank you for the interest on my implementation. The problem seems to be that the KL for the Gamma doesn't pull the latent space towards the prior, thus there is no regularization whatsoever. It could be a bug on the code, since I did not test any approache to mitigate this, like the beta-VAE or [KL annealing](https://arxiv.org/pdf/1511.06349.pdf), I am not really sure that everything is ok.

**Comment by AntixK** [comment]
Thanks for your input. I did try setting beta > 1 and it did not help. As you said, there isn't any regularization. that's happening during training. In fact, I had a tough time getting proper samples from Categorical VAE. I am currently working on this to see why this happens in Non-Gaussian VAEs in general - most probably a bug or I am missing something crucial!

P.S Scaling Gamma VAEs to larger datasets is a bit tricky! The loss almost always goes to nan/ -inf during training. I highly recommend gradient clipping and weight decay to stabilize the training.
